src/proxy/inbound.rs

```diff
@@ -402,8 +402,12 @@ impl Inbound {
             let from_gateway = match state.fetch_destination(&gateway_address.destination).await {
                 Some(address::Address::Workload(wl)) => Some(wl.identity()) == conn.src_identity,
                 Some(address::Address::Service(svc)) => {
-                    for (ip, _ep) in svc.endpoints.iter() {
-                        if state.fetch_workload(ip).await.map(|w| w.identity()) == conn.src_identity
+                    for (_ep_uid, ep) in svc.endpoints.iter() {
+                        if state
+                            .fetch_workload_by_uid(&ep.workload_uid)
```

Comment:
Interesting, so we fetch by UID because some don't have IP. Makes sense - maybe could use a comment inthe code, it wasn't obvious to me at first why.

Also I think this will never actually require a lookup _in current Istiod implementation_, as we push all Workloads when we push a Service. Good to have like this, though (and we had it before your PR as well, of course)